usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedTripTest.java

```diff
@@ -0,0 +1,219 @@
+        when(mockGenericObject.isSuccess()).thenReturn(false);
+        when(mockGenericObject.getData()).thenReturn(mockNoticeCollection);
+
+        //noinspection unchecked
+        when(mockBuilder.build()).thenReturn(mockGenericObject);
+
+        final ProcessParsedTrip underTest = new ProcessParsedTrip(mockResultRepo, mockGtfsDataRepo, mockBuilder);
+
+        when(mockParsedTrip.get("route_id")).thenReturn(null);
+        when(mockParsedTrip.get("service_id")).thenReturn("service_id");
+        when(mockParsedTrip.get("trip_id")).thenReturn("trip_id");
+        when(mockParsedTrip.get("trip_headsign")).thenReturn("trip_headsign");
+        when(mockParsedTrip.get("trip_short_name")).thenReturn("trip_short_name");
+        when(mockParsedTrip.get("direction_id")).thenReturn(1);
+        when(mockParsedTrip.get("block_id")).thenReturn("block_id");
+        when(mockParsedTrip.get("shape_id")).thenReturn("shape_id");
+        when(mockParsedTrip.get("wheelchair_accessible")).thenReturn(1);
+        when(mockParsedTrip.get("bikes_allowed")).thenReturn(1);
+
+        underTest.execute(mockParsedTrip);
+
+        verify(mockParsedTrip, times(1)).get(ArgumentMatchers.eq("route_id"));
+        verify(mockParsedTrip, times(1)).get(ArgumentMatchers.eq("service_id"));
+        verify(mockParsedTrip, times(1)).get(ArgumentMatchers.eq("trip_id"));
+        verify(mockParsedTrip, times(1)).get(ArgumentMatchers.eq("trip_headsign"));
+        verify(mockParsedTrip, times(1)).get(ArgumentMatchers.eq("trip_short_name"));
+        verify(mockParsedTrip, times(1)).get(ArgumentMatchers.eq("direction_id"));
+        verify(mockParsedTrip, times(1)).get(ArgumentMatchers.eq("block_id"));
+        verify(mockParsedTrip, times(1)).get(ArgumentMatchers.eq("shape_id"));
+        verify(mockParsedTrip, times(1)).get(ArgumentMatchers.eq("wheelchair_accessible"));
+        verify(mockParsedTrip, times(1)).get(ArgumentMatchers.eq("bikes_allowed"));
+
+        //noinspection ConstantConditions
+        verify(mockBuilder, times(1)).routeId(null);
+        verify(mockBuilder, times(1)).serviceId("service_id");
+        verify(mockBuilder, times(1)).tripId("trip_id");
+        verify(mockBuilder, times(1)).tripHeadsign("trip_headsign");
+        verify(mockBuilder, times(1)).tripShortName("trip_short_name");
+        verify(mockBuilder, times(1)).directionId(1);
+        verify(mockBuilder, times(1)).blockId("block_id");
+        verify(mockBuilder, times(1)).shapeId("shape_id");
+        verify(mockBuilder, times(1)).wheelchairAccessible(1);
+        verify(mockBuilder, times(1)).bikesAllowed(1);
+        verify(mockBuilder, times(1)).build();
+
+        verify(mockResultRepo, times(1)).addNotice(isA(Notice.class));
+        verifyNoMoreInteractions(mockParsedTrip, mockGtfsDataRepo, mockBuilder, mockResultRepo);
+    }
+
+    @Test
+    public void duplicateTripShouldAddNoticeToResultRepoAndShouldNotBeAddedToGtfsDataRepo() {
```

Comment:
It seems that is outside the scope of This specific use case tests. We test for duplication handling in GtfsDataRepository test class. Here, we shall only test the behavior when the repository refuses the add (returns null).
In consequence, I propose to 
- rename the previous method `invalidTripShouldAddNoticeToResultRepoAndShouldNotBeAddedToGtfsDataRepo` to `GtfsDataRepositoryAddFailureShouldAddNoticeToResultRepo`
- remove this test

Reply:
Ok I understand now that here we test the generation of the duplication notice, so my remark is null and void